test/e2e/annotations/auth.go

```diff
@@ -260,6 +260,32 @@ var _ = framework.IngressNginxDescribe("Annotations - Auth", func() {
 			})
 	})
 
+	When(`proxy-set-headers is set`, func() {
+		f := framework.NewDefaultFramework("auth-headers")
+
+		BeforeEach(func() {
+			f.NewEchoDeployment()
+		})
+
+		It(`should set "proxy_set_header 'My-Custom-Header' '42';"`, func() {
+			host := "auth"
+
+			annotations := map[string]string{
+				"nginx.ingress.kubernetes.io/auth-url":               "http://foo.bar/basic-auth/user/password",
+				"nginx.ingress.kubernetes.io/auth-proxy-set-headers": "auth-headers",
```

Comment:
@ElvinEfendi @aledbf This e2e test isn't working - it doesn't seem to be adding the external auth section to the nginx config at all. I've poked around the e2e framework and I'm not sure why this is failing but the snippet tests above succeed? Is there something obvious I'm missing?

Reply:
Are you sure `auth-headers` configmap gets created? Can you assert the existence of it here? Also why do you not create it programmatically like we do in other e2e tests.